src/backend/distributed/metadata/node_metadata.c

```diff
@@ -851,6 +852,17 @@ ActivateNode(char *nodeName, int nodePort)
 {
 	bool isActive = true;
 
+	/*
+	 * We currently require the object propagation to happen via superuser,
+	 * see #5139. While activating a node, we sync both metadata and object
+	 * propagation.
+	 *
+	 * In order to have a fully transactional semantics with add/activate
+	 * node operations, we require superuser. With that, we can guarantee
+	 * to send all commands within the same remote transaction.
+	 */
+	EnsureSuperUser();
```

Comment:
Are we adding this because of how we do our connection/transaction bookkeeping, or because we really need superuser?

Reply:
yes, that's the reason, clarified a bit further